src/music/win32_m.cpp

```diff
@@ -386,7 +386,7 @@ std::optional<std::string_view> MusicDriver_Win32::Start(const StringList &parm)
 
 				/* Compare requested and detected port name.
 				 * If multiple ports have the same name, this will select the last matching port, and the debug output will be confusing. */
-				if (portname != nullptr && strncmp(tryportname, portname, lengthof(tryportname)) == 0) port = tryport;
+				if (portname.has_value() && *portname == tryportname) port = tryport;
```

Comment:
The old code only did `starts_with`. Seems weird to me.

Reply:
It seemingly did, but didn't.

`convert_from_fs` ensures there's a `'\0'`. For this case, assume `tryportname` is full, so there is a `'\0'` at index 127. Given `strncmp` with 128, `tryportname` will always run into `'\0'`. If `portname` is longer (i.e. does not have `'\0'` at index 127), the strings are considered different.
So even though it looks like `strncmp`, I can't think of a way it could behave differently than a simple `strcmp`.

Reply:
Oh right, it uses `lengthof`, not the actual string length.